### PR TITLE
add parameters for PostgreSQL

### PR DESCRIPTION
**Pull request by sayaoailun**
add `conn_max_lifetime` and `conn_max_idle_time`

### CODE REVIEW

**Review by Vad1mo — APPROVED** [review]

**Comment by wy65701436** [comment]
@chlins we need to validate it before merging.

**Comment by chlins** [comment]
@sayaoailun Hi, please rebase your pr with main branch, thanks!

**Comment by sayaoailun** [comment]
> @sayaoailun Hi, please rebase your pr with main branch, thanks!

@chlins Done.

**Comment by chlins** [comment]
@sayaoailun Hi, thanks, but recently we've merged the [beego upgrade](https://github.com/goharbor/harbor/pull/17801) to the main which may have conflicts and impacts to your pr, so maybe you need to rebase again and then revise the code because beego orm update have some changes in this part.

**Comment by sayaoailun** [comment]
> @sayaoailun Hi, thanks, but recently we've merged the [beego upgrade](https://github.com/goharbor/harbor/pull/17801) to the main which may have conflicts and impacts to your pr, so maybe you need to rebase again and then revise the code because beego orm update have some changes in this part.

@chlins Done.

**Comment by chlins** [comment]
@sayaoailun Thanks your work!  I'll merge it after verification.

File: src/common/dao/pgsql.go
Comment (chlins) [inline src/common/dao/pgsql.go:104]: @sayaoailun Please review the code(line 98), there are some difference after beego upgrade.

Comment (sayaoailun) [reply src/common/dao/pgsql.go]: @chlins Fixed.

File: src/common/dao/pgsql.go
Comment (chlins) [inline src/common/dao/pgsql.go:103]: We'd better check error here to avoid db is nil then will panic in the next line.

Comment (sayaoailun) [reply src/common/dao/pgsql.go]: @chlins Fixed.

File: src/cmd/exporter/main.go
Comment (chlins) [inline src/cmd/exporter/main.go:27]: Print error log and should override the value to 5m if user input is invalid.

File: src/cmd/exporter/main.go
Comment (chlins) [inline src/cmd/exporter/main.go:31]: Print error log.

File: src/cmd/exporter/main.go
Comment (sayaoailun) [reply src/cmd/exporter/main.go]: @chlins Done.

File: src/cmd/exporter/main.go
Comment (sayaoailun) [reply src/cmd/exporter/main.go:33]: @chlins Done.

File: src/lib/config/metadata/value.go
Comment (chlins) [inline src/lib/config/metadata/value.go:148]: Please add UT coverage.

File: src/lib/config/metadata/metadatalist.go
Comment (chlins) [inline src/lib/config/metadata/metadatalist.go:107]: The default value should be "5m"?

File: src/lib/config/metadata/value.go
Comment (sayaoailun) [reply src/lib/config/metadata/value.go:148]: @chlins Done.

File: src/lib/config/metadata/metadatalist.go
Comment (sayaoailun) [reply src/lib/config/metadata/metadatalist.go]: @chlins Done.

**Review by chlins — APPROVED** [review]
lgtm

**Comment by chlins** [comment]
@sayaoailun Please check the CI action, the UTTEST failed.

**Comment by sayaoailun** [comment]
> @sayaoailun Please check the CI action, the UTTEST failed.

@chlins Fixed.

**Pull request merged**
